apptests/test_e2e.py

```diff
@@ -0,0 +1,117 @@
+import json
+import os
+import subprocess
+import time
+import unittest
+
+
+class TestYoutubeGenerator(unittest.TestCase):
+    def test_directory_and_file_creation(self):
+        command = (
+            "python3 generators/youtube_generator.py 10 -o ./testplans -c 19 -l en"
+        )
+        subprocess.run(command, shell=True)
+
+        target_directory = "./testplans"
+        self.assertTrue(
+            os.path.exists(target_directory), f"{target_directory} does not exist."
+        )
+        time_str = time.strftime("%Y%m%d-%H%M%S")
+
+        filename = f"Travel&Events_en_CAUQAQ_{time_str}.json"
+
+        json_file_path = os.path.join(target_directory, filename)
+        self.assertTrue(
+            os.path.exists(json_file_path), f"{json_file_path} does not exist."
+        )
+
+        with open(json_file_path, "r") as json_file:
+            try:
+                json.load(json_file)
+            except json.JSONDecodeError:
+                self.fail(f"{json_file_path} is not a valid JSON file.")
```

Comment:
I think this should create a folder for the test plan in the apptests directory, and then both the new directory and the test plan should be deleted as part of the cleanup.

Reply:
Dir for test generated testsplan is `apptests/testplans` and this dir is removed after "good path" test